src/components/action-popover/action-popover.spec.tsx

```diff
@@ -462,6 +447,7 @@ describe("ActionPopover", () => {
   });
 
   testStyledSystemMargin((props) => (
+    // ThemeProvider needs to remain here due to this test
```

Comment:
I understand why you added this comment, in the context of my previous comment about the use of `theme`, but in the context of the code itself (rather than the PR and its review comments) I don't think it adds anything and may be confusing. I think it's sufficiently obvious (at least to those who've worked on Carbon for a while) why a theme is needed for the styled system stuff to work correctly.

(Although maybe it's no so obvious after all - I tried the test without the ThemeProvider and you're right, they fail, but ideally we'd update the test utils so they work with the SageTheme... That's not relevant for this PR though! 😅 )